enums.rb

```diff
@@ -0,0 +1,136 @@
+# rubocop: disable Metrics/ModuleLength, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+# Description/Explanation of module enums
+module Enumerable
+  #:nodoc:
+  def my_each(index = 0)
+    return to_enum unless block_given?
+
+    while index < to_a.length
+      if instance_of?(Hash)
+        yield(keys[index], values[index])
+      else
+        yield(to_a[index])
+      end
+      index += 1
+    end
+    self
+  end
+
+  def my_each_with_index(index = 0)
+    return to_enum unless block_given?
+
+    while index < to_a.length
+      yield(to_a[index], index)
+      index += 1
+    end
+    self
+  end
+
+  def my_select(arr = [], index = 0)
+    return to_enum unless block_given?
+
+    my_each do |k|
+      if yield(k)
+        arr[i] = k
+        index += 1
+      end
+    end
+    arr
+  end
+
+  def check?(value, pattern, flag = nil)
+    if flag
+      return true if value.instance_of?(pattern)
+      return true if value.instance_of?(Complex) || value.instance_of?(Float) || value.instance_of?(Integer)
+    elsif value.to_s.match(pattern.to_s)
+      return true
+    end
+
+    false
+  end
+
+  def my_all?(pattern = nil, flag: false)
+    my_each do |k|
+      return false unless k
+
+      if block_given?
+        return false unless yield(k)
+      elsif (flag = [Integer, Float, String, Numeric].include?(pattern)) || pattern.to_s.include?('?')
+        return false unless check?(k, pattern, flag)
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil, flag: false)
+    my_each do |k|
+      if block_given?
+        return true if yield(k)
+      elsif (flag = [Integer, Float, String, Numeric].include?(pattern)) || pattern.to_s.include?('?')
+        return true if check?(k, pattern, flag)
+      elsif k
+        return true
+      end
+    end
+    false
+  end
```

Comment:
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```
  p %w[dog door rod blade].any?('test') # false
  p %w[dog door rod blade].my_any?('test') # true
```